app/ui/browser-modern/reducers/pages.js

```diff
@@ -186,6 +187,11 @@ function setPageUIState(state, pageId, pageUIState) {
   return state.withMutations(mut => {
     for (const [key, value] of Object.entries(pageUIState)) {
       mut.update('map', m => m.setIn([pageId, 'uiState', key], value));
+
+      if (key === 'pinned') {
+        const id = state.getIn(['map', pageId, 'uiState', 'webContentsId']);
+        ipcRenderer.send('guest-pinned-state', id, value);
```

Comment:
To elaborate, pinning a tab should become an effect, handled in the ui sagas, which sends this ipc event to the main process while also dispatching an action updating the local page state.